playground/infrastructure/helper.py

```diff
@@ -268,25 +270,28 @@ def _validate(tag: dict, supported_categories: List[str]) -> bool:
       In case tag is not valid, False
   """
   valid = True
+  tag_fields = {f.default for f in fields(TagFields)
+                if f.default not in
+                {o_f.default for o_f in fields(OptionalTagFields)}}
   # check that all fields exist and they have no empty value
-  for field in fields(TagFields):
-    if field.default not in tag:
+  for field in tag_fields:
+    if field not in tag:
       logging.error(
           "tag doesn't contain %s field: %s \n"
           "Please, check that this field exists in the beam playground tag."
           "If you are sure that this field exists in the tag"
           " check the format of indenting.",
-          field.default,
+          field,
           tag.__str__())
       valid = False
     if valid is True:
-      value = tag.get(field.default)
+      value = tag.get(field)
       if (value == "" or
-          value is None) and field.default != TagFields.pipeline_options:
+          value is None) and field != TagFields.pipeline_options:
         logging.error(
             "tag's value is incorrect: %s\n%s field can not be empty.",
             tag.__str__(),
-            field.default.__str__())
+            field.__str__())
```

Comment:
please use str cast instead of magic methods

Reply:
Done.